sw/device/lib/dif/dif_clkmgr.c

```diff
@@ -166,3 +173,139 @@ dif_result_t dif_clkmgr_external_clock_set_enabled(const dif_clkmgr_t *clkmgr,
                       extclk_ctrl_reg);
   return kDifOk;
 }
+
+dif_result_t dif_clkmgr_measure_ctrl_disable(const dif_clkmgr_t *clkmgr) {
+  if (clkmgr == NULL) {
+    return kDifBadArg;
+  }
+  mmio_region_write32(clkmgr->base_addr, CLKMGR_MEASURE_CTRL_REGWEN_REG_OFFSET,
+                      0);
+  return kDifOk;
+}
+
+dif_result_t dif_clkmgr_measure_ctrl_get_enable(const dif_clkmgr_t *clkmgr,
+                                                dif_toggle_t *state) {
+  if (clkmgr == NULL || state == NULL) {
+    return kDifBadArg;
+  }
+  *state = dif_bool_to_toggle(clkmgr_measure_ctrl_regwen(clkmgr));
+  return kDifOk;
+}
+
+dif_result_t dif_clkmgr_enable_measure_counts(const dif_clkmgr_t *clkmgr,
+                                              dif_clkmgr_measure_clock_t clock,
+                                              uint32_t lo_threshold,
+                                              uint32_t hi_threshold) {
+  if (clkmgr == NULL) {
+    return kDifBadArg;
+  }
+  if (!clkmgr_measure_ctrl_regwen(clkmgr)) {
+    return kDifLocked;
+  }
+
+  uint32_t reg_offset;
+  bitfield_bit32_index_t en_index;
+  bitfield_field32_t lo_field;
+  bitfield_field32_t hi_field;
+  switch (clock) {
+    case kDifClkmgrMeasureClockIo:
```

Comment:
I think the current popular way to deal with this is with a macro... like this:
```c
switch (block) {
  #define PICK_COUNT_CTRL_FIELDS(kind_) \
      reg_offset = CLKMGR_##kind_##MEAS__CTRL_SHADOWED_REG_OFFSET; \
     ... \
    break // No semicolon to force semicolons below.
     // etc.
  case kDifClkmgrMeasureClockIo:
    PICK_COUNT_CTRL_FIELDS(IO);
}
```

@timothytrippel might have thoughts on how to better use an X-macro here.

Reply:
Thanks for the suggestion. Done.